kernel/src/utilities/helpers.rs

```diff
@@ -42,3 +42,25 @@ macro_rules! count_expressions {
     ($head:expr $(,)?) => (1usize);
     ($head:expr, $($tail:expr),* $(,)?) => (1usize + count_expressions!($($tail),*));
 }
+
+/// Compute the CRC-32B checksum of a string.
+///
+/// Implementation based on https://lxp32.github.io/docs/a-simple-example-crc32-calculation/.
+/// Online calculator here: https://md5calc.com/hash/crc32b
+pub fn crc32b_str(s: &'static str) -> u32 {
```

Comment:
Might be nicer to have this accept a byte slice instead. This way we can use [`core:;str::as_bytes`](https://doc.rust-lang.org/stable/core/primitive.str.html#method.as_bytes) to calculate a CRC over a string, but also support other non-unicode byte sequences. Shouldn't need a lot of changes, just iterating over the bytes and removing the cast.